File: practice/regex/applications/find_a_substring/FindSubstring.scala
import scala.io.Source

object FindSubstring extends App {
  val inputLines = Source.stdin.getLines().toList
  val linesLength = inputLines(0).toInt
  val lines = inputLines.slice(1, linesLength + 1)
  val substrings = inputLines.slice(linesLength + 2, inputLines.length)
  val counts = substrings.map(findSubstring(lines, _))
  println(counts.mkString("\n"))


  def findSubstring(lines: List[String], substring: String): Int = {
    lines.map(line => {
      val pattern = "\\w+" + substring + "\\w+"
      val matches = pattern.r.findAllMatchIn(line)
      matches.length
    }).sum
  }
}

File: practice/regex/applications/find_a_substring/FindSubstring.js
'use strict';

const processData = input => {
    let inputLines = input.split('\n');
    let linesLength = parseInt(inputLines[0]);
    let substrings = inputLines.slice(linesLength + 2);
    for(let j = 0; j < substrings.length; j++) {
        let count = 0;
        let pattern = "\\w+" + substrings[j] + "\\w+";
        for(let i = 1; i <= linesLength; i++) {
            let line = inputLines[i];
            let match = line.match(new RegExp(pattern, "g"));
            count += match ? match.length : 0;
        }
        console.log(count);
    }
};

process.stdin.resume();
process.stdin.setEncoding("ascii");

var _input = "";
process.stdin.on("data", input => _input += input);
process.stdin.on("end", () => processData(_input));